### DISCUSSION TITLE
Unique serial number on Raspberry Pi Pico

### DISCUSSION THREAD

**Discussion by robzed** [tabemann/zeptoforth#45] [Q&A] [answered]
Anyone got any idea how I'd access this from Zeptoforth?

https://www.raspberrypi.com/documentation/pico-sdk/group__pico__unique__id.html

Background is that I have a radio chip that requires per device calibration - the radio chip itself doesn't have an unique ID and neither does the RP2040 ... but the flash has a unique ID.

**Comment by tabemann** [comment]
Currently zeptoforth has no access to this (I was not aware of the existence of this until you mentioned it today). I would probably have to engineer in code to talk to the flash chip to get this info in the kernel (it would have to be in the kernel because only the kernel is guaranteed to run from RAM only).

**Comment by pkoning2** [comment]
This feature was added in release 1.4.0.

**Reply by tabemann** [reply, marked as answer]
It is now implemented as `unique-id`, which returns a 64-bit value.
